src/CMakeLists.txt

```diff
@@ -1291,11 +1291,12 @@ if( UNIX )
 	endif()
 endif()
 
-target_link_libraries( zdoom ${PROJECT_LIBRARIES} gdtoa lzma ${ZMUSIC_LIBRARIES} )
+target_link_libraries( zdoom ${PROJECT_LIBRARIES} gdtoa lzma ${ZMUSIC_LIBRARIES} gvizdoom )
 #TODO opencv temp
 target_include_directories( zdoom PUBLIC ${OpenCV_INCLUDE_DIRS} )
 #TODO end of opencv temp
 
+target_include_directories( gvizdoom PUBLIC ../include/gvizdoom )
```

Comment:
My philosophy regarding these is usually that if innermost directories are added they are listed as `PRIVATE` properties, so that files can be directly included only in the library itself (here of course the distinction is a little bit shady since contents of all subfolders are used in the same target, but let's think each directory as a separate "library" for now). Then as a `PUBLIC` property I'd usually add the top-level include directory, so that the library from which the header is included from is indicated by the `#include` directive (for example, `gvizdoom/Action.hpp`). This also allows files with identical names to be present without introducing ambiguity to includes.
In this case, `../include/` is already present in include directories so I'd say that this could be removed and includes of form `#include <gvizdoom/Header.hpp>` be used in files outside the `gvizdoom/` directory.